### ISSUE TITLE
Параллельное бронирование

### ISSUE THREAD

**Issue by Warheart666** [savely03/booking_app#15]
С каким поведением мы столкнемся, если сейчас два пользователя одновременно начнут бронировать один номер на одну дату?
как выйти из сложившейся ситуации?

**Comment by savely03** [comment]
Да, если будет параллельное бронирование, я сделал метод findFreeRoomByHotelAndDate в roomRepositoy с pessimistic_write блокировкой, то есть этот метод ищет свободный номер в отеле на определенные даты и на уровне базы блокирует строку, остальные пользователи, которые хотят вызвать данный метод будут в ожидании. Но тут нужно устанавливать по идее timeout, поэтому в bookingService, в методе создания бронирования стоит timeout на уровне транзакции.
